util: add DestinationChip and DestinationInputChip components

tony-tvu opened the pull request:
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
This pull request introduces the DestinationChip and DestinationInputChip components, an addition that is essential for facilitating the development of generic destinations with pluggable providers. These components play a role in processing props such as icons, text, and cancel functions, and rendering them on the screen.

**Which issue(s) this PR fixes:**
Part of https://github.com/target/goalert/issues/2639

**Screenshots:**
<img width="1062" alt="chips" src="https://github.com/target/goalert/assets/52386267/afd43ad0-039a-40d7-b44d-b71bf32e1687">

Icon Variations:
<img width="1108" alt="icon variations" src="https://github.com/target/goalert/assets/52386267/f93719dc-8cf2-41df-8be4-a6376dacdf81">

**Additional Info:**
Run `make storybook` to interact with new components.

These components will be unused in main application functions, until we have wired them in with future component updates, e.g. PolicyStepDialog components.

Forfold commented:
Could you pls add to the description screenshots with the intended UX and the info for using the experimental flag?

tony-tvu commented:
> Could you pls add to the description screenshots with the intended UX and the info for setup (experimental flag, mutation to have Dest types added, etc)?

I updated the description to include a screenshot and added additional context. This won't need any experiment flags to operate.

ethan-haynes reviewed (dismissed):
LGTM 👍

Forfold commented on web/src/app/util/DestinationChip.tsx:
```diff
@@ -0,0 +1,115 @@
+            <CircularProgress size='1em' />
+          </Avatar>
+        }
+        label='loading...'
```
```suggestion
        label='Loading...'
```

Forfold commented on web/src/app/util/DestinationChip.tsx:
```diff
@@ -0,0 +1,115 @@
+        }
+        label='loading...'
+        onDelete={
+          props.onDelete
+            ? (e) => {
+                e.stopPropagation()
+                e.preventDefault()
+                props.onDelete?.()
+              }
+            : undefined
```
I've been told not to use undefined explicitly, so I will do the same here

```suggestion
          props.onDelete
            ? (e) => {
                e.stopPropagation()
                e.preventDefault()
                props.onDelete?.()
              }
            : null
```

tony-tvu replied:
I'm afraid this is not an option because the mui chips component expects the `onDelete` prop to be a function or undefined.
<img width="656" alt="Screenshot 2024-01-10 at 10 15 51 AM" src="https://github.com/target/goalert/assets/52386267/bd110cc4-4f17-459e-816f-ac49c49d898d">

Forfold replied:
I agree! Just going off of precedent.

Forfold commented on web/src/app/util/DestinationChip.tsx:
```diff
@@ -0,0 +1,115 @@
+              e.preventDefault()
+              props.onDelete?.()
+            }
+          : undefined
```
same

Forfold commented on web/src/app/util/DestinationInputChip.stories.tsx:
```diff
@@ -0,0 +1,75 @@
+  },
+} satisfies Meta<typeof DestinationInputChip>
+
+export default meta
```
can this be supplied with the declaration?

export default const meta = {

Forfold replied:
Weird. What error?

allending313 reviewed (approved):
lgtm!
